src/world/shipparts/missile_launcher.rs

```diff
@@ -0,0 +1,80 @@
+use crate::timer::Timer;
+use crate::world::types::{Location, Module, ModuleInputs, WorldEvent};
+use mlua::prelude::{LuaTable, LuaValue};
+use mlua::{Lua, Nil, Result, ToLua, UserData, UserDataMethods};
+
+struct MissileLauncher {
+    charged: bool,
+    recharge_timer: Timer,
+}
+
+impl MissileLauncher {
+    fn process(&self, orders: Vec<String>) -> bool {
+        orders.iter().any(|order| order == "shoot")
+    }
+}
+
+impl Module for MissileLauncher {
+    fn update(&mut self, inputs: ModuleInputs, location: Location) -> Option<WorldEvent> {
+        if !self.charged {
+            if self.recharge_timer.ready(inputs.dt) {
+                self.charged = true;
+            }
+            None
+        } else if inputs.controls.gun {
+            // Check if there is a part one block in front of the gun.
+            let part = match inputs.get_part.call::<_, LuaValue>((location, [0, 1])) {
```

Comment:
This is our first example of calling Lua code from Rust. Currently the missile_launcher code has to be aware that it's Lua, but it could be wrapped in a native Rust function.